docs/features/incremental-generators.cookbook.md

```diff
@@ -0,0 +1,656 @@
+            fullyQualifiedMetadataName: "GeneratedNamespace.GeneratedAttribute",
+            predicate: static (syntaxNode, cancellationToken) => syntaxNode is BaseMethodDeclarationSyntax,
+            transform: static (context, cancellationToken) =>
+            {
+                var containingClass = context.TargetSymbol.ContainingType;
+                return new Model(
+                    // Note: this is a simplified example. You will also need to handle the case where the type is in a global namespace, nested, etc.
+                    Namespace: containingClass.ContainingNamespace?.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat),
+                    ClassName: containingClass.Name,
+                    MethodName: context.TargetSymbol.Name);
+            }
+        );
+
+        // Register a factory that can create our custom syntax receiver
+        context.RegisterSourceOutput(pipeline, static (context, model) =>
+        {
+            var sourceText = SourceText.From($$"""
+                namespace {{model.Namespace}};
+                partial class {{model.ClassName}}
+                {
+                    partial void {{model.MethodName}}()
+                    {
+                        // generated code
+                    }
+                }
+                """, Encoding.UTF8);
+
+            context.AddSource($"{model.ClassName}_{model.MethodName}.g.cs", sourceText);
+        });
+    }
+
+    private record Model(string Namespace, string ClassName, string MethodName);
+}
+```
+
+### Issue Diagnostics
+
+**User Scenario:** As a generator author I want to be able to add diagnostics to the user's compilation.
+
+**Solution:** We do not recommend issuing diagnostics within generators. It is possible, but doing so without breaking incrementality is advanced topic
+beyond the scope of this cookbook. Instead, we suggest writing a separate analyzer for reporting diagnostics.
+
+### INotifyPropertyChanged
+
+**User scenario:** As a generator author I want to be able to implement the `INotifyPropertyChanged` pattern automatically for a user.
+
+**Solution:** The design tenant 'Explicitly additive only' seems to be at direct odds with the ability to implement this, and appears to call for user code modification.
+However we can instead take advantage of explicit fields and instead of *editing* the users properties, directly provide them for listed fields.
+
+**Example:**
+
+Given a user class such as:
+
+```csharp
+using AutoNotify;
+
+public partial class UserClass
+{
+    [AutoNotify]
+    private bool _boolProp;
+
+    [AutoNotify(PropertyName = "Count")]
+    private int _intProp;
+}
+```
+
+A generator could produce the following:
+
+```csharp
+using System;
+using System.ComponentModel;
+
+namespace AutoNotify
+{
+    [AttributeUsage(AttributeTargets.Field, Inherited = false, AllowMultiple = false)]
+    sealed class AutoNotifyAttribute : Attribute
```

Comment:
Your docs are a bit consistent with how it generates attributes into code: 

- `public` and unsealed
- `internal` (implicitly) and `sealed`

Think the doc should be consistent in how it emits attributes. My preference is `internal sealed` but I wouldn't push back on other forms.